src/bindings/python/src/pyopenvino/graph/discrete_type_info.cpp

```diff
@@ -14,10 +14,32 @@
 
 namespace py = pybind11;
 
+// DiscreteTypeInfo doesn't own provided memory. Wrapper allows to avoid leaks.
+class DiscreteTypeInfoWrapper : public ov::DiscreteTypeInfo {
+private:
+    const std::string name_str;
+    const std::string version_id_str;
+
+public:
+    DiscreteTypeInfoWrapper(std::string _name_str, std::string _version_id_str)
+        : DiscreteTypeInfo(nullptr, nullptr, nullptr),
+          name_str(std::move(_name_str)),
+          version_id_str(std::move(_version_id_str)) {
+        name = name_str.c_str();
+        version_id = version_id_str.c_str();
+    }
+};
+
 void regclass_graph_DiscreteTypeInfo(py::module m) {
     py::class_<ov::DiscreteTypeInfo, std::shared_ptr<ov::DiscreteTypeInfo>> discrete_type_info(m, "DiscreteTypeInfo");
```

Comment:
```suggestion
    py::class_<DiscreteTypeInfoWrapper, ov::DiscreteTypeInfo, std::shared_ptr<DiscreteTypeInfoWrapper>> discrete_type_info(m, "DiscreteTypeInfo");
```

Shouldn't it be exposed like this so `DiscreteTypeInfoWrapper` is recognized but also inherits from DiscreteTypeInfo in case it needs to be returned? I mean something is missing here either one should be exposed and wrapped in all places (like InferRequestWrapper) or create both classes.

Reply:
Discussed offline -- seems fine to expose it like this.